File: 02/dotnet6/magnitude.cs
var path = System.Reflection.Assembly.GetExecutingAssembly().Location;
string[] command = System.IO.File.ReadAllLines($"{Path.GetDirectoryName(path)}\\input.txt");

// Part 1
int horiz = 0, depth = 0;
foreach (string c in command)
{
    var parts = c.Split(' ');
    int magnitude = int.Parse(parts[1]);
    switch (parts[0])
    {
        case "forward":
            horiz += magnitude;
            break;
        case "backward":
            horiz -= magnitude;
            break;
        case "down":
            depth += magnitude;
            break;
        case "up":
            depth -= magnitude;
            break;
    }
}
Console.WriteLine($"Depth {depth} pos {horiz} - {horiz * depth}");

// Part 2
horiz = 0;
depth = 0;
int aim = 0;
foreach (string c in command)
{
    var parts = c.Split(' ');
    int magnitude = int.Parse(parts[1]);
    switch (parts[0])
    {
        case "forward":
            horiz += magnitude;
            depth += (aim * magnitude);
            break;
        case "backward":
            horiz -= magnitude;
            depth -= (aim * magnitude);
            break;
        case "down":
            aim += magnitude;
            break;
        case "up":
            aim -= magnitude;
            break;
    }
}
Console.WriteLine($"Depth {depth} pos {horiz} - {horiz * depth}");

File: 02/python/magnitude.py
import os
file_path = os.path.dirname(__file__)
with open("%s\\input.txt" % file_path) as f:
    lines = f.readlines()

horiz = 0
depth = 0
for command in lines:
    parts = command.split(' ')
    magnitude = int(parts[1])
    if parts[0] == "forward":
        horiz += magnitude
    if parts[0] == "backward":
        horiz -= magnitude
    if parts[0] == "down":
        depth += magnitude
    if parts[0] == "up":
        depth -= magnitude
print("Depth %d pos %d - %d" % (depth, horiz, depth * horiz))

# Part 2
horiz = 0
depth = 0
aim = 0
for command in lines:
    parts = command.split(' ')
    magnitude = int(parts[1])
    if parts[0] == "forward":
        horiz += magnitude
        depth += (aim * magnitude)
    if parts[0] == "backward":
        horiz -= magnitude
        depth -= (aim * magnitude)
    if parts[0] == "down":
        aim += magnitude
    if parts[0] == "up":
        aim -= magnitude
print("Depth %d pos %d - %d" % (depth, horiz, depth * horiz))
